### DISCUSSION TITLE
I don't understand how it work

### DISCUSSION THREAD

**Discussion by VladNVN** [wazuh/wazuh#24700] [Q&A]
Event:
<13>1 2024-07-09T09:42:41.440809+00:00 xxx.yyy ulogd - - [NXLOG@12345 timestamp="2024-07-09 09:42:41" timestamp_str="2024-07-09T12:42:41.440809+03:00" sensor_id="139772" event_type="firewall" src_ip="111.111.111.111" src_port="54486" dest_ip="222.222.222.222" dest_port="443" proto="TCP" action="allowed" signature_id="186" signature="443/TCP" category="Firewall" msgrepeatcount="0" revision="0" dest_country="private" src_country="private" username="" EventReceivedTime="2024-07-09 09:42:46" SourceModuleName="zzz_100.100.100.100" SourceModuleType="im_dbi"]

Decoders:
![image](https://github.com/user-attachments/assets/e87bf10e-515e-4a0b-8b09-f8f6776d4417)

Full regex continent-firewall: src_ip="\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}"\ssrc_port="\d{1,}"\sdest_ip="\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}"\sdest_port="\d{1,}"\sproto="\w{1,}"\saction="\w{1,}"

wazuh-logtest output:

**Phase 1: Completed pre-decoding.
        full event: '<13>1 2024-07-09T09:42:41.440809+00:00 xxx.yyy ulogd - - [NXLOG@12345 timestamp="2024-07-09 09:42:41" timestamp_str="2024-07-09T12:42:41.440809+03:00" sensor_id="139772" event_type="firewall" src_ip="111.111.111.111" src_port="54486" dest_ip="222.222.222.222" dest_port="443" proto="TCP" action="allowed" signature_id="186" signature="443/TCP" category="Firewall" msgrepeatcount="0" revision="0" dest_country="private" src_country="private" username="" EventReceivedTime="2024-07-09 09:42:46" SourceModuleName="zzz_100.100.100.100" SourceModuleType="im_dbi"]'

**Phase 2: Completed decoding.
        name: 'continent-syslog'

Questions:
1. Why wazuh don't parse timestamp?
2. Why wazuh don't use second decoder?
3. How fix it?

**Comment by VladNVN** [comment]
I understand!

**Reply by SvetGeor-sg** [reply]
I have a similar problem, I am ingesting some logs from Trend Micro and they have 3 fields with date/time - one is the rsyslog timestamp, the other is the generated timestamp and the third is the received timestamp.
I want the generated timestamp to be the one acknowledge during the pre-decoding. 

I have tried using parent and child decoders to simplify the regex but didn't work.

here is the log example
2024-08-21T16:30:02.574632+03:00 Rsyslogsrv TrendMicro <116>1 2024-08-21 16:24:13 192.168.0.10 WFBS-SVC-AC [LogUrlFiltering@1.3.6.1.4.1.6101 Action="Access to a website was blocked" Device name="device" Domain="OFFICE" Generated="2024-08-21T16:20:00+03:00" Group name="Device (Default)" IPv6 Address="-" Label="" Received="2024-08-21T16:24:13+03:00" URL="https://someurl/init2.0.js" URL Category="Spam" User="some.user"]


Please help! :)
